### ISSUE TITLE
Outputs reference variables instead of resources created by this module

### ISSUE THREAD

**Issue by nottooshort** [lacework/terraform-aws-cloudtrail#44]
Output:
https://github.com/lacework/terraform-aws-cloudtrail/blob/4c66775acbfad026e2a70f7211a21f7598006613/output.tf#L36-L39

Variable (referenced in output):
https://github.com/lacework/terraform-aws-cloudtrail/blob/4c66775acbfad026e2a70f7211a21f7598006613/variables.tf#L25-L29

Usage:
```
module "aws_cloudtrail" {
  source  = "lacework/cloudtrail/aws"
  version = "~> 0.1"

  use_existing_iam_role   = false
  use_existing_cloudtrail = true
  bucket_name             = aws_s3_bucket.global_cloudtrail_storage.id
  bucket_arn              = aws_s3_bucket.global_cloudtrail_storage.arn
  bucket_sse_key_arn      = aws_kms_key.cloudtrail_key.arn
}

data "aws_iam_policy_document" "lacework_kms_decrypt" {
  statement {
    effect = "Allow"
    actions = [
      "kms:Decrypt"
    ]
    resources = [aws_kms_key.cloudtrail_key.arn]
  }
}

resource "aws_iam_policy" "lacework_kms_policy" {
  name        = "lacework-kms-decryption"
  path        = "/"
  description = "Supplimental policy to allow lacework to decrypt log results"
  policy      = data.aws_iam_policy_document.lacework_kms_decrypt.json
}

resource "aws_iam_policy_attachment" "lacework_kms_attach" {
  name       = "lacework-kms-decryption"
  roles      = [module.aws_cloudtrail.iam_role_name]
  policy_arn = aws_iam_policy.lacework_kms_policy.arn
}
```

The block of code above returns a null value for module.aws_cloudtrail.iam_role_name. We need to attach a kms decrypt policy to allow the iam role created by this module, however the outputs of this module do not reference the resources it creates because the outputs are simply the inputs.

Terraform Plan:
```
Terraform will perform the following actions:

  # aws_iam_policy_attachment.lacework_kms_attach will be created
  + resource "aws_iam_policy_attachment" "lacework_kms_attach" {
      + id         = (known after apply)
      + name       = "lacework-kms-decryption"
      + policy_arn = "arn:aws:iam::123456789012:policy/lacework-kms-decryption"
      + roles      = [
          + "",
        ]
    }

Plan: 1 to add, 0 to change, 0 to destroy.
```

Terraform Apply:
```
undefinedInitializing plugins and modules...
aws_iam_policy_attachment.lacework_kms_attach: Creating...

Error: No Users, Roles, or Groups specified for IAM Policy Attachment lacework-kms-decryption

  on lacework.tf line 29, in resource "aws_iam_policy_attachment" "lacework_kms_attach":
  29: resource "aws_iam_policy_attachment" "lacework_kms_attach" {
```

**Comment by nottooshort** [comment]
The IAM resource created by this module:
module.aws_cloudtrail.module.lacework_ct_iam_role.aws_iam_role.lacework_iam_role[0]

This resource is not callable to return the ARN, Name, etc.

It would be helpful for the outputs to reference the resource instead of locals or other variables.

**Comment by afiune** [comment]
The fix is available in version [v0.2.2](https://github.com/lacework/terraform-aws-cloudtrail/releases/tag/v0.2.2)
